code/WorkInProgress/parallax.dm

```diff
@@ -0,0 +1,214 @@
+			parallax_settings += new /datum/parallax_object/planet/iudicium
+			parallax_settings += new /datum/parallax_object/ss14
+			parallax_settings += new /datum/parallax_object/ss12
+			parallax_settings += new /datum/parallax_object/planet/iustitia
+			parallax_settings += new /datum/parallax_object/planet/magus
+			parallax_settings += new /datum/parallax_object/planet/mundus
+			parallax_settings += new /datum/parallax_object/star/blue
+			parallax_settings += new /datum/parallax_object/star
+			parallax_settings += new /datum/parallax_object/planet/x3
+
+
+			// generate parallax_objects list out of settings
+			for(var/datum/parallax_object/P as anything in parallax_settings)
+				var/centerx = round((PARALLAX_VIEW_WIDTH - 4)/2)
+				var/centery = round((PARALLAX_VIEW_HEIGHT - 4)/2)
+				var/atom/movable/screen/hud/N = create_screen(null,null, P.icon,P.icon_state, "[centerx],[centery]", HUD_LAYER)
+				N.plane = PLANE_SPACE
+				N.layer = HUD_LAYER + P.layer
+				N.appearance_flags += TILE_BOUND
+				N.mouse_opacity = 0
+
+				parallax_coords[N] = P.coords
+				scale[N] = P.scale
+				size[N] = P.size
+				parallax_zlevel[N] = P.zlevel
+
+				parallax_objects[N] = N
+				N.transform = matrix(0,0,0,0,0,0)
+
+	proc/update(var/turf/master_turf) // hopefully this is fast enough
+		if(!active) return
+
+		for(var/atom/movable/screen/hud/P as anything in parallax_objects)
+
+			if(parallax_zlevel[P] != master_turf.z)
+				P.transform = matrix(0,0,0,0,0,0)
+				continue
+			else
+				var/matrix = matrix(size[P],0,(parallax_coords[P]["X"]-master_turf.x)*(scale[P])*32-icon(P.icon,P.icon_state).Width()/2,0, size[P],(parallax_coords[P]["Y"]-master_turf.y)*(scale[P])*32-icon(P.icon,P.icon_state).Height()/2)
+				animate(P,transform=matrix,time=round(world.icon_size/master.glide_size*world.tick_lag,world.tick_lag))
+				continue
+
+		// background scrolling
+		background.transform = matrix(master?.client.view ?2.1:1.5,0,(150-master_turf.x)/2,0,1.5,(150-master_turf.y)/2)
+
+		//animate(background,transform=matrix,time=round(world.icon_size/master.glide_size*world.tick_lag,world.tick_lag))
+
+	proc/toggle() // makes it so update() doesnt have to loop through extra stuff
+		var/setting = master?.client?.parallax
+		if(!setting)
+			for(var/atom/movable/screen/hud/P as anything in parallax_objects)
+				P.transform = matrix(0,0,0,0,0,0)
+			active = FALSE
+		else
+			active = TRUE
+
+
+// below stuff is everything that will trigger the update() proc
+/mob
+	OnMove()
+		..()
+		if(!isnull(src.client))
+			var/turf/NewLoc = get_turf(src)
+			src.parallax.update(NewLoc)
+		for(var/mob/M as anything in src.observers)
+			if(isnull(M.client))
+				continue
+			var/turf/NewLoc = get_turf(M)
+			M.parallax.update(NewLoc)
+/obj/machinery/vehicle
+	Move(var/NewLoc)
+		..()
+		for(var/mob/M as anything in src.contents)
+			if(!isnull(M.client))
+				M.parallax.update(NewLoc)
+			for(var/mob/Mo as anything in M.observers)
```

Comment:
why is this nested
ditto for obj/vehicle

Reply:
main idea i had with this was to make it update for anyone observing the mob in the vehicles

Reply:
you do know the first loop will runtime right - did you test this?
`for(var/mob/M as anything in src.contents)`

Reply:
didn't test it at the time, its been since replaced with a signal